pkg/kubelet/images/image_manager.go

```diff
@@ -167,6 +252,24 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, pod *v1.Pod, conta
 	m.logIt(ref, v1.EventTypeNormal, events.PulledImage, logPrefix, fmt.Sprintf("Successfully pulled image %q in %v (%v including waiting)",
 		container.Image, imagePullResult.pullDuration.Truncate(time.Millisecond), time.Since(startTime).Truncate(time.Millisecond)), klog.Info)
 	m.backOff.GC()
+
+	if m.state != nil {
+		m.lock.Lock()
```

Comment:
if mutations to m.lock are guarded by mutex already, do we still need `m.lock.Lock()`?

Reply:
Removed.